### ISSUE TITLE
Update docker file Fedora 33  to F35 as Fedora 33 is nearing EOL

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/installer#397]
Update redhat and suse docker file Fedora 33  to F35 as Fedora 33 is nearing EOL

**Comment by sophia-guo** [comment]
Migrate from Fedora 33 to 35 and got :

```
Error: Failed to download metadata for repo 'fedora': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64 [getaddrinfo() thread failed to start]
```
Docker version 20.10.6, build 370c289
ubuntu version: Description:	Ubuntu 16.04.7 LTS

**Comment by sophia-guo** [comment]
Similar to https://medium.com/nttlabs/ubuntu-21-10-and-fedora-35-do-not-work-on-docker-20-10-9-1cd439d9921. Tried with workaround, which will hit other issues in the following steps.

```
+ rpmdev-setuptree
error: Failed to open shell expansion pipe for command: echo $HOME: Cannot allocate memory 
error: Failed to open shell expansion pipe for command: echo $HOME: Cannot allocate memory 
```

**Comment by jerboaa** [comment]
Wouldn't the fix be to upgrade to docker 20.10.10 and better? Fedora 35's `moby-engine` is at `20.10.11`.

**Comment by sophia-guo** [comment]
Believe there are only three machines labeled `docker&&linux&&x64` [https://ci.adoptopenjdk.net/label/docker&&linux&&x64/] .  what's the docker version and could we update to docker 20.10.10 to see if this problem could be fixed?

**Comment by sophia-guo** [comment]
@sxa I'm do the tests on 147.75.204.239, which Docker version is 20.10.6

**Comment by sophia-guo** [comment]
updated the docker on my local mac, docker version is 20.10.11. Works fine. Issue opened in https://github.com/adoptium/infrastructure/issues/2408

**Comment by sophia-guo** [comment]
Interesting this doesn't happen to  https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-amd-1/, which has the same docker version as https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-intel-1/ and Dynamic docker machines from Azure. Not sure what's the configuration differences between https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-amd-1/ and https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-intel-1/ or Dyanmic docker machines from Azure. @sxa

**Comment by sxa** [comment]
> Interesting this doesn't happen to https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-amd-1/, which has the same docker version as https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-intel-1/ and Dynamic docker machines from Azure. Not sure what's the configuration differences between https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-amd-1/ and https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-intel-1/ or Dyanmic docker machines from Azure. @sxa

I know George had logged in and some something to the docker config on the AMD box a few weeks ago, so I'll see if I can unpick that and find out why they're different ... But I don't plan to do anything to the machine mentioned in my previous comment.

**Comment by sxa** [comment]
Docker service restarted on [docker-package-ubuntu2004-intel-1](https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-intel-1/) and that appears to have resolved the issue on that host.

**Comment by sophia-guo** [comment]
> I did the tests on 147.75.204.239, which Docker version is 20.10.6

Yes, that is not the machine without label `dockerBuild`, but turns out quite similar to the machines labeled with `dockerBuild` and I do have access to that machine. I have no direct access to machines with label `dockerBuild`( Accurately , not for the majority of the machines), which is painful when working on some new stuff. So for me the process is all the installer stuff are tested on 147.75.204.239 first and then verified by jenkins job on machines labeled with `dockerBuild`. ( Two reasons i don't like to run the tests on my local is one is too slow, the second is it may be different from jenkins ci).

**Comment by sophia-guo** [comment]
@sxa is it possible to restart docker service on 147.75.204.239 (test-packet-ubuntu1604-x64-1), which has same docker version as other two `dockerBuild` machine and has the same issue with fedora:35? If the issue could also be fixed we may come to a conclusion this issue can be fixed by restart the docker service for docker version 20.10.6?

**Comment by sophia-guo** [comment]
On my local restart doesn't work. Only update the docker version works. Don't know the really underlying reason for this.

**Comment by sxa** [comment]
Why is the process trying to use test-packet-ubuntu1604-x64-1? It doesn't have the dockerBuild label on it so there shouldn't be a problem with it not working for Fedora35

**Comment by sophia-guo** [comment]
@gdams what's the configuration difference between the Azure dynamic machines and other two machines labeled with `dockerBuild` - https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-amd-1/ and https://ci.adoptopenjdk.net/computer/docker-packet-ubuntu2004-intel-1/ ?

When updating to use Fedora:35 Azure dynamic machines will get the following error:
https://ci.adoptopenjdk.net/view/work-in-progress/job/Sophia-adoptium-packages-linux-pipeline/56/consoleFull

```
Error: Failed to download metadata for repo 'fedora': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64 [getaddrinfo() thread failed to start]
```
The other two machines either don't have this issue or the issue can be fixed by restart docker service.
 
My local environment has the same issue and can be fixed by update docker version. Not sure what is the underlying issue could you please take a look?

**Comment by sophia-guo** [comment]
> Why is the process trying to use test-packet-ubuntu1604-x64-1? It doesn't have the dockerBuild label on it so there shouldn't be a problem with it not working for Fedora35

The reason I'm using test-packet-ubuntu1604-x64-1 is [(https://github.com/adoptium/installer/issues/397#issuecomment-992601864)](https://github.com/adoptium/installer/issues/397#issuecomment-992601864)

It happened to have the same issue as other machines. By fix it we probably can come to conclusion the issue can be fixed by restart the docker service for those similar machines. https://github.com/adoptium/installer/issues/397#issuecomment-992630681

**Comment by sxa** [comment]
>  is it possible to restart docker service on 147.75.204.239 (test-packet-ubuntu1604-x64-1)
I've restarted. It makes no difference. The failure still occurs. I would STRONGLY suggest that we stop testing anything on that machine and move you to one of the others.

**Comment by sxa** [comment]
Infra issue raised at https://github.com/adoptium/infrastructure/issues/2410 and access granted :-)
